modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -462,13 +463,43 @@ def merge(self, right, **kwargs):
             def map_func(left, right=right, kwargs=kwargs):
                 return pandas.merge(left, right, **kwargs)
 
+            new_columns = None
+            new_dtypes = None
+            if self._modin_frame._columns_cache is not None:
+                if left_on is None and right_on is None:
+                    if on is None:
+                        on = [c for c in self.columns if c in right.columns]
```

Comment:
Shouldn't we check that `right._modin_frame._columns_cache is not None`?

Reply:
`right` is a materialized pandas dataframe here, so we shouldn't
https://github.com/modin-project/modin/blob/62be5ae8ee74dcd5d2703379872ae46bac936177/modin/core/storage_formats/pandas/query_compiler.py#L457